torch/nn/modules/module.py

```diff
@@ -729,14 +729,47 @@ def share_memory(self):
     def _get_name(self):
         return self.__class__.__name__
 
-    def __repr__(self):
-        tmpstr = self._get_name() + '(\n'
+    def get_extra_repr(self):
+        r"""Get the extra representation of the module
+
+        To print customized extra information, you should reimplement
+        this method in your own modules. Each item in the returned list will
+        be printed in separate line.
+        """
+        return []
+
+    def _get_extra_lines(self):
+        r"""Helper method to process extra representation"""
+        extra_repr = self.get_extra_repr()
+        if isinstance(extra_repr, str):
+            return extra_repr.split('\n')
+        else:
+            return extra_repr
+
+    def _get_child_lines(self):
+        child_lines = []
         for key, module in self._modules.items():
-            modstr = module.__repr__()
-            modstr = _addindent(modstr, 2)
-            tmpstr = tmpstr + '  (' + key + '): ' + modstr + '\n'
-        tmpstr = tmpstr + ')'
-        return tmpstr
+            mod_str = module.__repr__()
+            mod_str = _addindent(mod_str, 2)
+            child_lines.append('(' + key + '): ' + mod_str)
+        return child_lines
```

Comment:
This is not getting called anywhere outside of `__repr__`, so let's keep it there

Reply:
Speaking of `__repr__`, I think `repr(module)` is more pythonic than `module.__repr__()`.